src/jit/core_templates.expr

```diff
@@ -987,6 +987,46 @@
       (carg $0 ptr)
       (carg (const (&QUOTE MVM_reg_obj) int_sz) int))))
 
+(template: unshift_i!
+  (callv (^getf (^repr $0) MVMREPROps pos_funcs.unshift)
+    (arglist
+      (carg (tc) ptr)
+      (carg (^stable $0) ptr)
+      (carg $0 ptr)
+      (carg (^body $0) ptr)
+      (carg $1 ptr)
+      (carg (const (&QUOTE MVM_reg_int64) int_sz) int))))
+
+(template: unshift_n!
+  (callv (^getf (^repr $0) MVMREPROps pos_funcs.unshift)
+    (arglist
+      (carg (tc) ptr)
+      (carg (^stable $0) ptr)
+      (carg $0 ptr)
+      (carg (^body $0) ptr)
+      (carg $1 ptr)
+      (carg (const (&QUOTE MVM_reg_num64) int_sz) int))))
+
+(template: unshift_s!
+  (callv (^getf (^repr $0) MVMREPROps pos_funcs.unshift)
+    (arglist
+      (carg (tc) ptr)
+      (carg (^stable $0) ptr)
+      (carg $0 ptr)
+      (carg (^body $0) ptr)
+      (carg $1 ptr)
+      (carg (const (&QUOTE MVM_reg_str) int_sz) int))))
+
+(template: unshift_o!
+  (callv (^getf (^repr $0) MVMREPROps pos_funcs.unshift)
+    (arglist
+      (carg (tc) ptr)
+      (carg (^stable $0) ptr)
+      (carg $0 ptr)
+      (carg (^body $0) ptr)
+      (carg $1 ptr)
+      (carg (const (&QUOTE MVM_reg_obj) int_sz) int))))
```

Comment:
I don't think the `!` means, what you think it means, and given that I've implemented it, I'm probably right :wink: 
The `!` means that the template writes a value to a register (has a write operand), but doesn't the expression doesn't yield a value but writes directly to the output register. It means that the template is *destructive*, hence the `!` was chosen for similarity with the scheme convention.
Since none of the unshift operators actually have an output operand, the `!` is wrong here, and after I've merged #950 it will error out, too.

Reply:
Thanks. I've updated the ops to remove the '!'.